packages/grid/src/GridUtils.ts

```diff
@@ -745,90 +745,107 @@ export class GridUtils {
     visibleIndex: VisibleIndex,
     movedItems: MoveOperation[] = []
   ): ModelIndex {
-    let modelIndex = visibleIndex;
-
-    for (let i = movedItems.length - 1; i >= 0; i -= 1) {
-      const movedItem = movedItems[i];
-      if (modelIndex === movedItem.to) {
-        ({ from: modelIndex } = movedItem);
-      } else if (movedItem.from <= modelIndex && modelIndex < movedItem.to) {
-        modelIndex += 1;
-      } else if (movedItem.to < modelIndex && modelIndex <= movedItem.from) {
-        modelIndex -= 1;
-      }
-    }
+    const modelIndex = GridUtils.applyItemMoves(
+      visibleIndex,
+      visibleIndex,
+      movedItems,
+      true
+    )[0][0];
 
     return modelIndex;
   }
 
   /**
-   * Translate the provided UI start/end indexes to the model start/end indexes by applying the `movedItems` transformations.
-   * Since moved items can split apart a range, multiple pairs of indexes are returned
-   *
-   * @param start Start item in one dimension
-   * @param end End item in one dimension
-   * @param movedItems Moved item pairs in this dimension
-   * @returns Array of start/end pairs of the indexes after transformations applied.
+   * Applies the items moves to the 1D range
+   * @param start The start index of the range
+   * @param end The end index of the range
+   * @param movedItems The move operations to apply
+   * @param reverse If the moved items should be applied in reverse (this reverses the effects of the moves)
+   * @returns A list of 1D ranges in the translated space. Possibly multiple non-continuous ranges
    */
-  static getModelRangeIndexes(
+  private static applyItemMoves(
+    start: number,
+    end: number,
+    movedItems: MoveOperation[],
+    reverse?: boolean
+  ): [number, number][];
+
+  // Without this duplicate signature, TS will complain
+  private static applyItemMoves(
     start: GridRangeIndex,
     end: GridRangeIndex,
-    movedItems: MoveOperation[]
+    movedItems: MoveOperation[],
+    reverse?: boolean
+  ): GridAxisRange[];
+
+  private static applyItemMoves(
+    start: GridRangeIndex,
+    end: GridRangeIndex,
+    movedItems: MoveOperation[],
+    reverse = false
   ): GridAxisRange[] {
+    if (typeof start === 'string') {
+      return [];
+    }
     if (start == null || end == null) {
       return [[start, end]];
     }
```

Comment:
Should add a test case for `null`s... as I think this is incorrect. If the range is `[5, null]` and we have move `[1, 10]`, then I would expect it to be `[[6, 9], [1, 1], [10, null]]`... (`null` meaning selecting to the end of that range).